src/evict/evict_inline.h

```diff
@@ -588,30 +588,44 @@ __wti_evict_hs_dirty(WT_SESSION_IMPL *session)
       ((uint64_t)(conn->evict->eviction_dirty_trigger * bytes_max) / 100));
 }
 
+/*
+ * __evict_check_user_ok_with_eviction --
+ *     Check if the user wants to abort eviction in this session. #private
+ */
+static bool
+__evict_check_user_ok_with_eviction(WT_SESSION_IMPL *session, bool busy)
+{
+    /* Ask the user only if eviction is optional and it's a user session. */
+    if (busy || F_ISSET(session, WT_SESSION_INTERNAL))
+        return (true);
+
+    WT_EVENT_HANDLER *event_handler = session->event_handler;
+    if (event_handler->handle_general != NULL &&
+      event_handler->handle_general(
+        event_handler, &S2C(session)->iface, &session->iface, WT_EVENT_EVICTION, NULL) != 0)
+        return (false);
+
+    return (true);
+}
+
 /* !!!
  * __wt_evict_app_assist_worker_check --
  *     Check if eviction trigger thresholds have reached to determine whether application threads
  *     should assist eviction worker threads with eviction of pages from the queues.
  *
  *     Input parameters:
- *       (1) `busy`: A flag indicating if the session is actively pinning resources, in which
- *            case dirty trigger is ignored.
+ *       (1) `busy`: A flag indicating if eviction is mandatory (true) or optional (false).
  *       (2) `readonly`: A flag indicating if the session is read-only, in which case dirty and
  *            update triggers are ignored.
  *       (3) `didworkp`: A pointer to indicate whether eviction work was done (optional).
  *
- *     Return an  error code from `__wti_evict_app_assist_worker` if it is unable to perform
+ *     Return an error code from `__wti_evict_app_assist_worker` if it is unable to perform
  *     meaningful work (eviction cache stuck).
  */
 static WT_INLINE int
 __wt_evict_app_assist_worker_check(
   WT_SESSION_IMPL *session, bool busy, bool readonly, bool *didworkp)
 {
-    WT_BTREE *btree;
```

Comment:
I wish we could do this as part of a seperate PR. It seems like it's not part of the scope of this PR. It could be distracting when I come back to this PR and see other unrelated changes.

Reply:
Also, this causes inconsistencies with other functions that do not follow this pattern, do we have a ticket for that?